src/HomeWork4/HomeWork4.java

```diff
@@ -0,0 +1,180 @@
+        System.out.println("");
+        /*
+        * Task #19
+        * Имеется строка с текстом. Подсчитать количество слов в тексте. Желательно учесть, что слова могут разделяться
+        * несколькими пробелами, в начале и конце
+        * текста также могут быть пробелы, но могут и отсутствовать.
+        *
+        * */
+        System.out.println("Task #19");
+        String text19 = "Имеется строка с текстом. Подсчитать количество слов в тексте. Желательно\n" +
+                "учесть, что слова могут разделяться несколькими пробелами, в начале и конце\n" +
+                "текста также могут быть пробелы, но могут и отсутствовать.";
+        int countText19 = 0;
+        String [] strings = text19.split("\\s+");
+        for (int i = 0; i < strings.length - 1; i++){
+            String text19C = strings[i];
+            if (text19C != null){
+                countText19++;
+            }
+        }
+        System.out.println("Всего слов в тексте : " + countText19);
+        System.out.println("");
+        /*
+        * Task #20
+        * Имеется строка с текстом. Вывести текст, составленный из последних бук всех слов.
+        *
+        * */
+        System.out.println("Task #20");
+        String text20 ="Вывести текст, составленный из последних бук всех слов";
+        StringBuilder builder = new StringBuilder();
+        Pattern pattern = Pattern.compile("([а-яА-Я]+)");
+        Matcher matcher = pattern.matcher(text20);
+        int count20 = 0;
+        while (matcher.find()){
+            builder.append(text20.substring(matcher.end()-1,matcher.end()));
+            count20++;
+        }
+        System.out.println("Количество слов в тексте : " + count20 + "\nПослелние буквы в словах : " + builder);
+        System.out.println("");
+        /*
+        * Task #21
+        *Напишите три цикла выполняющих многократное сложение строк, один с
+        * помощью оператора сложения и String, другой с помощью StringBuilder и метода
+        * append, а также аналогино для StringBuffer. Сравните скорость их выполнения.
+        * */
+        System.out.println("Task #21");
+        String text21 = "Напишите три цикла выполняющих многократное сложение строк, один с\n" +
+                "помощью оператора сложения и String, другой с помощью StringBuilder и метода\n" +
+                "append, а также аналогино для StringBuffer. Сравните скорость их выполнения.\n" +
+                "Регулярные выражения";
+        String[] text21Strings = text21.split("\\s+");
+        String text21S = "";
+        long start= System.nanoTime();
+
+
+        for (int i = 0; i< text21Strings.length; i++){
+            text21S += text21Strings[i];
+        }
+
+        long end =  System.nanoTime();
+        long result = end - start;
+
+        System.out.println("Добавление строк String . \nВремени затрачено на добавление :" + result / 1000 +
+                " мс. \nРезультат доьвыления : " + text21S );
```

Comment:
all good, but it is not milliseconds. for example you can concatenate a string multiple times